### PR TITLE
Bf2 workflow

### PR DESCRIPTION
**Pull request by CPage-Ten10**

### CODE REVIEW

File: src/test/resources/features/complaints/ComplaintsTriage.feature [hunk 1]
```diff
@@ -309,7 +309,7 @@ Feature: Complaints Triage
     And I submit details on the Triage Capture Reason page
     And I send the case to drafting
     Then the case should be moved to the "DRAFT" stage
-    And the summary should display the owning team as "Border Force"
+    And the summary should display the owning team as "Border Force (Stage 2)"
```
Comment (CaseyProsser) [inline src/test/resources/features/complaints/ComplaintsTriage.feature:312]: Think this is a mistake

File: src/test/resources/features/complaints/ComplaintsTriage.feature [hunk 2]
```diff
@@ -319,7 +319,7 @@ Feature: Complaints Triage
     And I submit details on the Triage Capture Reason page
     And I escalate the case to WFM at Triage stage
     Then the case should be moved to the "Escalated to WFM" stage
-    And the summary should display the owning team as "Border Force"
+    And the summary should display the owning team as "Border Force (Stage 2)"
```
Comment (CaseyProsser) [inline src/test/resources/features/complaints/ComplaintsTriage.feature:322]: Mistake here?

File: src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java [hunk 3]
```diff
@@ -293,17 +293,17 @@ public void iSearchForTheCaseByItsCaseReference() {
         }
     }
 
-    @And("I search for the COMP case escalated to COMP2 by it's case reference")
+    @And("I search for the complaints case escalated to stage 2 by it's case reference")
     public void iSearchForTheEscalatedCOMPCaseByCaseReference() {
-        String compCaseRef = sessionVariableCalled("compCaseReference");
-        search.enterComplaintsSearchCriteria("Case Reference", compCaseRef);
+        String complaintCaseRef = sessionVariableCalled("complaintsCaseReference");
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java:298]: Can we set this session variable to be a bit more descriptive? Something like "stage2ComplaintsCaseReference" would be good

File: src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java [hunk 4]
```diff
@@ -293,17 +293,17 @@ public void iSearchForTheCaseByItsCaseReference() {
...
+        search.enterComplaintsSearchCriteria("Case Reference", complaintCaseRef);
         safeClickOn(searchButton);
         search.waitForResultsPage();
     }
 
-    @And("I load the COMP2 case by selecting its case reference from the Escalate Case column")
+    @And("I load the stage 2 complaints case by selecting its case reference from the Escalate Case column")
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java:304]: could you change this to "I load the stage 2 UKVI complaint case by......"

File: src/test/java/com/hocs/test/pages/complaints/BFProgressCase.java [hunk 5]
```diff
@@ -60,6 +65,37 @@ public void createCaseOfTypeAndMoveItToTargetStageWithSpecifiedComplaintType(Str
...
 
+    public void attemptEscalateBFCaseToStage2() throws Exception {
+        dashboard.selectSearchLinkFromMenuBar();
+        search.enterComplaintsSearchCriteria("Complainant Home Office Reference", getCurrentMonth() + "/" + getCurrentYear());
+        search.clickTheButton("Search");
+        search.waitForResultsPage();
+        if (search.checkVisibilityOfEscalationHypertext()) {
+            WebElementFacade bfCaseRefField = findBy("//a[contains(text(), 'Escalate case')]/parent::td/preceding-sibling::td/a");
+            String bfCaseRef = bfCaseRefField.getText();
+            setSessionVariable("complaintsCaseReference").to(bfCaseRef);
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/pages/complaints/BFProgressCase.java:76]: could we change this method variable and session variable to indicate which of the 2 case references this is, the stage1 or stage 2 reference.

File: src/test/java/com/hocs/test/pages/complaints/COMPProgressCase.java [hunk 6]
```diff
@@ -106,9 +106,9 @@ public void attemptEscalateCOMPCaseToStage2() throws Exception {
         if (search.checkVisibilityOfEscalationHypertext()) {
             WebElementFacade compCaseRefField = findBy("//a[contains(text(), 'Escalate case')]/parent::td/preceding-sibling::td/a");
             String compCaseRef = compCaseRefField.getText();
-            setSessionVariable("compCaseReference").to(compCaseRef);
+            setSessionVariable("complaintsCaseReference").to(compCaseRef);
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/pages/complaints/COMPProgressCase.java:109]: could we change this to indicate which stage this reference is from, stage 1 or stage 2 case?

File: src/test/java/com/hocs/test/pages/decs/Search.java [hunk 7]
```diff
@@ -962,7 +962,7 @@ public boolean zeroSearchResultsReturned() {
         return getNumberOfSearchResults() == 0;
     }
 
-    public void selectCOMP2CaseRefOfEscalatedCOMPCase(String stage1CaseRef) {
+    public void selectComplaintsStage2CaseRefOfEscalatedCOMPCase(String stage1CaseRef) {
         WebElementFacade comp2CaseRef = findBy("//a[text()='" + stage1CaseRef + "']/parent::td/following-sibling::td/a");
```
Comment (CaseyProsser) [inline src/test/java/com/hocs/test/pages/decs/Search.java:966]: This variable and half of the method name still are for COMP, did we need to label this for use in all complaints escalations?

File: src/test/resources/features/complaints/ComplaintsTriage.feature (hunk 1 above)
Comment (CPage-Ten10) [reply src/test/resources/features/complaints/ComplaintsTriage.feature]: Yeah, I accidentally changed the wrong scenarios when I realised stage 2 had it's own team

File: src/test/resources/features/complaints/ComplaintsTriage.feature (hunk 2 above)
Comment (CPage-Ten10) [reply src/test/resources/features/complaints/ComplaintsTriage.feature]: Same as above, resolved

File: src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java (hunk 3 above)
Comment (CPage-Ten10) [reply src/test/java/com/hocs/test/glue/decs/SearchStepDefs.java]: Updated to stage2CaseReference

**Review by CaseyProsser — APPROVED** [review]

**Pull request merged**